examples/05-sage-list-compose-fleet/index.ts

```diff
@@ -0,0 +1,142 @@
+        connection,
+        sageProgram as any,
+        Starbase,
+        'confirmed',
+        [
+            {
+                memcmp: {
+                    offset: 8 + 1,
+                    bytes: sageGameIdPubkey.toBase58(),
+                },
+            },
+            {
+                memcmp: {
+                    offset: 8 + 1 + 32,
+                    bytes: x58,
+                },
+            },
+            {
+                memcmp: {
+                    offset: 8 + 1 + 32 + 8,
+                    bytes: y58,
+                },
+            },
+        ],
+    );
+
+    if (starbase === undefined || starbase.type !== 'ok') {
+        throw Error('no starbase found');
+    }
+
+    const [starbasePlayer] = await readAllFromRPC(
+        connection,
+        sageProgram as any,
+        StarbasePlayer,
+        'confirmed',
+        [
+            {
+                memcmp: {
+                    offset: 8 + 1,
+                    bytes: playerPofilePubkey.toBase58(),
+                },
+            },
+            {
+                memcmp: {
+                    offset: 8 + 1 + 32,
+                    bytes: sageGameIdPubkey.toBase58(),
+                },
+            },
+            {
+                memcmp: {
+                    offset: 8 + 1 + 32 + 32,
+                    bytes: starbase.key.toBase58(),
+                },
+            },
+        ],
+    );
+
+    if (starbasePlayer === undefined || starbasePlayer.type !== 'ok') {
+        throw Error('no starbase player found');
+    }
+
+    // console.log(JSON.stringify(starbase, null, 2));
+    const starbaseData = starbase.data.data;
+    const starbaseName = byteArrayToString(starbaseData.name);
+
+    // console.log(JSON.stringify(starbasePlayer, null, 2));
+    const starbasePlayerData = starbasePlayer.data.data;
+    const shipEscrowCount = starbasePlayerData.shipEscrowCount;
+
+    console.log(`Starbase: ${starbaseName} (${starbase.key.toBase58()})`);
+    console.log(`Player: ${playerPofilePubkey.toBase58()}`);
+    console.log(`Ship Escrow Count: ${shipEscrowCount}`);
+
+    for (let i = 0; i < starbasePlayer.data.wrappedShipEscrows.length; i++) {
+        const entry = starbasePlayer.data.wrappedShipEscrows[i];
+
+        const ship = await readFromRPCOrError(
+            connection,
+            sageProgram as any,
+            entry.ship,
+            Ship,
+            'confirmed'
+        );
+
+        const shipData = ship.data;
+        const shipName = byteArrayToString(shipData.name);
+
+        console.log(`Ship: ${shipName} (${entry.amount}) Mint: ${shipData.mint.toBase58()}`); 5
+    }
+}
+
+// cd examples
+// bun run 05-sage-list-compose-fleet/index.ts 40 30
+const wallet = process.env.WALLET_PATH || "/home/user/.config/solana/id.json"
+const rpcEndpoint = process.env.RPC_ENDPOINT || "https://mainnet.helius-rpc.com/?api-key=";
+const rpcWebsocket = process.env.RPC_WEBSOCKET || "wss://rpc.helius.xyz/?api-key=";
```

Comment:
How about defaulting to a public RPC?

Reply:
@mindrunner sure we change those to something that makes more sense but in my examples I've been following the standard set by @cuniculture in his first example.

https://github.com/staratlasmeta/star-atlas-cookbook/blob/master/sage-player-profile/index.ts#L10C1-L11C55

If we make that change I would purpose that it be a sweeping change across the entire project not just the PR.